### PR TITLE
Launches switching ASNetworkImageNode callbacks to global queue.

### PR DESCRIPTION
**Pull request by garrettmoon**
This showed performance gains for us on startup when the main thread was under heavy contention.

### CODE REVIEW

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/559a1a2baad0750dbb1806eade7ead38/raw/51c09f11acf5ca0df2442c39831553fe1290de62/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/6f34b82434b1276afb5713b8b26a994e/raw/2e7c6b82ae2648c6c0687e8264c85af29931c0d4/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/6cc69b2f283b0e0dcc4cc5f405a74277/raw/8afd0b050ea05bc845901742a4add5e7b0a3f72f/log.txt)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/6082d75f8ef91891d9aeb266848728ba/raw/94103d2aaa6a1f89418594af7e23f6b5fc8955b6/log.txt)

File: Tests/ASNetworkImageNodeTests.mm [hunk 1]
```diff
@@ -109,7 +109,6 @@ @implementation ASTestImageCache
 
 - (void)cachedImageWithURL:(NSURL *)URL callbackQueue:(dispatch_queue_t)callbackQueue completion:(ASImageCacherCompletion)completion
 {
-  ASDisplayNodeAssert(callbackQueue == dispatch_get_main_queue(), @"ASTestImageCache expects main queue for callback.");
```
Comment (garrettmoon) [inline Tests/ASNetworkImageNodeTests.mm]: @Adlai-Holler do you remember why this was here?

Comment (Adlai-Holler) [reply Tests/ASNetworkImageNodeTests.mm]: Nope, we can remove.

**Pull request merged**